xbmc/cores/VideoPlayer/DVDSubtitles/DVDSubtitleTagSami.cpp

```diff
@@ -14,6 +14,56 @@
 #include "utils/RegExp.h"
 #include "utils/StringUtils.h"
 
+#include <algorithm>
+#include <map>
+
+namespace
+{
+// W3C HTML 16 basic colors
+const std::map<std::string, std::string> HTMLCOLORS = {
+    {"white", "FFFFFF"}, {"silver", "C0C0C0"}, {"gray", "808080"},    {"black", "000000"},
+    {"red", "FF0000"},   {"maroon", "800000"}, {"yellow", "FFFF00"},  {"olive", "808000"},
+    {"lime", "00FF00"},  {"green", "008000"},  {"aqua", "00FFFF"},    {"teal", "008080"},
+    {"blue", "0000FF"},  {"navy", "000080"},   {"fuchsia", "FF00FF"}, {"purple", "800080"}};
+
```

Comment:
Reading the European Broadcasting Union EBU‑TT/TTML  spec, cyan and magenta are missing in this list:
See page 44 and 45 of the document on https://tech.ebu.ch/publications/tech3350
So, adding cyan and magenta will fix all subtitle colors in Kodi add-ons from European broadcasters:
{"cyan", "00FFFF"}, {"magenta","FF00FF"}

Reply:
the intent for now is to support basic HTML colors and so not include colors of other format types
with future rework i will change the list with the CSS color list that include all HTML colors